### PR TITLE
MNTOR-2566: send status code 401 for unauth'd API endpoints instead of redirect

### PR DESCRIPTION
**Pull request by mansaj**
# References:

Jira: MNTOR-2566

# Description
Send 401s for APIs. No redirect for API endpoints

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Review by rhelmer — APPROVED** [review]

File: src/app/api/v1/hibp/notify/route.ts [hunk 1]
```diff
@@ -108,7 +108,7 @@ export async function POST(req: NextRequest) {
     pubsub = new PubSub({ projectId });
   } catch (ex) {
     logger.error("Error connecting to PubSub:", ex);
-    return NextResponse.json({ success: false }, { status: 500 });
+    return NextResponse.json({ success: false }, { status: 429 });
```
Comment (Vinnl) [inline src/app/api/v1/hibp/notify/route.ts:111]: Cool, first time I'm seeing 429 Too Many Requests in use! So I assume this is usually a rate-limiting error?

**Review by Vinnl — APPROVED** [review]
Thanks!

File: src/app/api/v1/hibp/notify/route.ts (hunk 1 above)
Comment (rhelmer) [reply src/app/api/v1/hibp/notify/route.ts:111]: Yes that's right, currently we alert based on HTTP 500s (from the loadbalancer logs) so 429 is a more useful signal both for us and for the caller.

Comment (rhelmer) [reply src/app/api/v1/hibp/notify/route.ts:111]: If we want to get fancier with it we can include a https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After header as well, and implement actual rate limiting.

**Pull request merged**